test/main/Test/Calendar/HasDays.hs

```diff
@@ -0,0 +1,112 @@
+module Test.Calendar.HasDays (
+    testHasDays,
+) where
+
+import Data.Time.Calendar
+import Data.Time.Calendar.Month
+import Data.Time.Calendar.Quarter
+import Test.Tasty
+import Test.Tasty.HUnit
+import Test.Tasty.QuickCheck
+
+newtype WDay = MkWDay Day
+    deriving (Eq, Show)
+
+instance Arbitrary WDay where
+    arbitrary = do
+        (MkWYear y) <- arbitrary
+        (MkWMonthOfYear m) <- arbitrary
+        (MkWDayOfMonth d) <- arbitrary
+        pure $ MkWDay $ YearMonthDay y m d
+
+newtype WYear = MkWYear Year
+    deriving (Eq, Show)
+
+instance Arbitrary WYear where
+    arbitrary = fmap MkWYear $ choose (-1000, 3000)
```

Comment:
Some of these wrappers `W` already exist in other tests, however, I didn't want to add too much noise in the context of this PR since reusing them requires some additional refactoring.